pkg/kubelet/dockershim/docker_image.go

```diff
@@ -34,7 +34,7 @@ func (ds *dockerService) ListImages(filter *runtimeapi.ImageFilter) ([]*runtimea
 	opts := dockertypes.ImageListOptions{}
 	if filter != nil {
 		if imgSpec := filter.GetImage(); imgSpec != nil {
-			opts.MatchName = imgSpec.Image
+			opts.Filters.Add("reference", imgSpec.Image)
```

Comment:
Followed the implementation at https://github.com/moby/moby/blob/master/client/image_list.go#L21-L23. I guess we want to add a test for filter in ListImage in a separate PR?